File: ShowDown_test/HumanPlayer.js
const readline = require('readline');
const Player = require('./Player');

class HumanPlayer extends Player{
    constructor(){
        super();
        this.rl = readline.createInterface({
            input: process.stdin,
            output: process.stdout
        });
    }

    nameHimself(){
        this.rl.question(`Enter name for Human Player: `, (name) => {
            this.name = name;
            this.rl.close();
            callback();
        });
    }
}
module.exports = HumanPlayer; 

File: ShowDown_Java/HumanPlayer.java
package ShowDown_Java;

import java.util.Scanner;

public class HumanPlayer extends Player implements CommandLine{

    // constructor 
    public HumanPlayer() {
        
    }

    @Override 
    public void nameHimself() {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Please enter your name: ");
        String playerName = scanner.nextLine();
        this.name = playerName;
        // System.out.println("Your name is set to: " + this.name);
        scanner.close();
    }
}
